cleanlab/rank.py

```diff
@@ -556,52 +554,73 @@ def get_confidence_weighted_entropy_for_each_label(
     return label_quality_scores
 
 
-def get_knn_distance_ood_scores(
-    features: np.ndarray, nbrs: NearestNeighbors, k: int = None
+def get_outlier_scores(
+    features: np.ndarray, knn: NearestNeighbors = None, k: int = 10
 ) -> np.ndarray:
-    """Returns the KNN distance out-of-distribution (OOD) score for each datapoint.
+    """Returns the KNN distance outlier score for each example.
 
-    This is a function to compute OOD scores where higher scores indicate the datapoint is more likely to be OOD.
+    Returns an outlier score for each example based on its feature values, where higher scores indicate examples more
+    likely to be outliers. The score is based on the average distance between the example and its
+    K nearest neighbors in the dataset (in feature space).
 
     Parameters
     ----------
-    features : np.array
-      Feature matrix of shape (N, M), where N is the number of datapoints and M is the number of features.
-      This is the "query set" of features for each datapoint which are used for nearest neighbor search.
+    features : np.ndarray
+      Feature array of shape ``(N, M)``, where N is the number of examples and M is the number of features used to represent each example.
+      All features should be numeric. For unstructured data (eg. images, text, categorical values, ...), you should provide
+      vector embeddings to represent each example (e.g. extracted from some pretrained neural network).
 
-    nbrs : sklearn.neighbors.NearestNeighbors
-      Instantiated NearestNeighbors class object that's been fitted on a dataset in the same feature space.
+    knn : sklearn.neighbors.NearestNeighbors, default = None
+      Instantiated ``NearestNeighbors`` class object that's been fitted on a dataset in the same feature space.
       Note that the distance metric and n_neighbors is specified when instantiating this class.
+      You can also pass in a subclass of ``sklearn.neighbors.NearestNeighbors`` which allows you to use faster
+      approximate neighbor libraries as long as you wrap them behind the same sklearn API.
+      If you specify ``knn`` here and wish to find outliers in the same data you already passed into ``knn.fit(features)``, you should specify ``features = None`` here if your ``knn.kneighbors(None)``
+      returns the distances to the datapoints it was ``fit()`` on.
```

Comment:
I found the above sentence confusing. Is the user ever supposed to specify `features=None`?

Reply:
Either features or knn can be ``None`` but both cannot be ``None`` at the same time. I like the idea of going with two Optional arguments and then adding a runtime check to make sure at least one of the values is not ``None``.